app/repositories/SessionRepository.scala

```diff
@@ -133,10 +132,12 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
       Left(DatabaseError)
     }
 
-  def getGuidanceSession(key: String, processCode: String): Future[RequestOutcome[Session]] =
+  def getGuidanceSession(key: String, processCode: String, requestId: Option[String]): Future[RequestOutcome[Session]] =
     findAndUpdate(
       Json.obj("_id" -> key),
-      Json.obj(List(toFieldPair("$set", Json.obj(toFieldPair(TtlExpiryFieldName, Json.obj("$date" -> Instant.now().toEpochMilli))))).toArray: _*),
+      Json.obj("$set" -> Json.obj((List(toFieldPair(TtlExpiryFieldName, Json.obj(toFieldPair("$date", Instant.now().toEpochMilli)))) ++
+                                        requestId.toList.map(rId => toFieldPair(RequestId, rId))).toArray: _*)),
+      //Json.obj(List(toFieldPair("$set", Json.obj(toFieldPair(TtlExpiryFieldName, Json.obj("$date" -> Instant.now().toEpochMilli))))).toArray: _*),
```

Comment:
Are we holding on to this commented out code for a reason?

Reply:
removed now and pushed to branch